core/src/trezor/ui/layouts/tt.py

```diff
@@ -0,0 +1,410 @@
+    text.bold(prompt)
+    text.br()
+    text.br_half()
+    text.normal("By continuing you agree")
+    text.br()
+    text.normal("to")
+    text.bold("https://trezor.io/tos")
+    return interact(
+        ctx,
+        Confirm(text, major_confirm=True),
+        "setup_device",
+        ButtonRequestType.ResetDevice,
+    )
+
+
+async def confirm_backup(ctx: wire.GenericContext) -> bool:
+    text1 = Text("Success", ui.ICON_CONFIRM, ui.GREEN)
+    text1.bold("New wallet created", "successfully!")
+    text1.br_half()
+    text1.normal("You should back up your", "new wallet right now.")
+
+    text2 = Text("Warning", ui.ICON_WRONG, ui.RED)
+    text2.bold("Are you sure you want", "to skip the backup?")
+    text2.br_half()
+    text2.normal("You can back up your", "Trezor once, at any time.")
+
+    if (
+        await interact(
+            ctx,
+            Confirm(text1, cancel="Skip", confirm="Back up", major_confirm=True),
+            "backup_device",
+            ButtonRequestType.ResetDevice,
+        )
+        is CONFIRMED
+    ):
+        return True
+
+    confirmed = (
+        await interact(
+            ctx,
+            Confirm(text2, cancel="Skip", confirm="Back up", major_confirm=True),
+            "backup_device",
+            ButtonRequestType.ResetDevice,
+        )
+    ) is CONFIRMED
+    return confirmed
+
+
+def confirm_path_warning(ctx: wire.GenericContext, path: str) -> LayoutType:
+    text = Text("Confirm path", ui.ICON_WRONG, ui.RED)
+    text.normal("Path")
+    text.mono(*break_path_to_lines(path, MONO_CHARS_PER_LINE))
+    text.normal("is unknown.", "Are you sure?")
+    return interact(
+        ctx,
+        Confirm(text),
+        "path_warning",
+        ButtonRequestType.UnknownDerivationPath,
+    )
+
+
+def _show_qr(
+    address: str,
+    desc: str,
+    cancel: str = "Address",
+) -> Confirm:
+    QR_COEF = const(4) if len(address) < QR_SIZE_THRESHOLD else const(3)
+    qr = Qr(address, QR_X, QR_Y, QR_COEF)
+    text = Text(desc, ui.ICON_RECEIVE, ui.GREEN)
+
+    return Confirm(Container(qr, text), cancel=cancel, cancel_style=ButtonDefault)
+
+
+def _split_address(address: str) -> Iterator[str]:
+    return chunks(address, MONO_CHARS_PER_LINE)
+
+
+def _hex_lines(hex_data: str, lines: int = TEXT_MAX_LINES) -> Iterator[str]:
+    if len(hex_data) >= MONO_HEX_PER_LINE * lines:
+        hex_data = hex_data[: (MONO_HEX_PER_LINE * lines - 3)] + "..."
+    return chunks(hex_data, MONO_HEX_PER_LINE)
+
+
+def _show_address(
+    address: str,
+    desc: str,
+    network: str = None,
+) -> Confirm:
+    text = Text(desc, ui.ICON_RECEIVE, ui.GREEN)
+    if network is not None:
+        text.normal("%s network" % network)
+    text.mono(*_split_address(address))
+
+    return Confirm(text, cancel="QR", cancel_style=ButtonDefault)
+
+
+def _show_xpub(xpub: str, desc: str, cancel: str) -> Paginated:
+    pages = []  # type: List[ui.Component]
```

Comment:
please use the new-style type annotation

Reply:
Fixed in 379440b0a6baa70f6bebf4542e3fdee21c6f092c.